src/contracts/custom.ts

```diff
@@ -0,0 +1,61 @@
+import { ContractMetadata } from "../core/classes/contract-metadata";
+import { IStorage, NetworkOrSignerOrProvider } from "../core";
+import { SDKOptions } from "../schema/sdk-options";
+import { ContractWrapper } from "../core/classes/contract-wrapper";
+import {
+  IThirdwebContract,
+  IThirdwebContract__factory,
+} from "@thirdweb-dev/contracts";
+import { CustomContractSchema } from "../schema/contracts/custom";
+import { UpdateableNetwork } from "../core/interfaces/contract";
+
+/**
+ *
+ * @internal
+ */
+export class CustomContract implements UpdateableNetwork {
+  static contractType = "custom" as const;
+  static schema = CustomContractSchema;
+
+  private contractWrapper;
+  public metadata;
+  private storage;
+
+  constructor(
+    network: NetworkOrSignerOrProvider,
+    address: string,
+    storage: IStorage,
+    options: SDKOptions = {},
+    abi: any = undefined,
+    contractWrapper = new ContractWrapper<IThirdwebContract>(
+      network,
+      address,
+      abi || IThirdwebContract__factory.abi,
+      options,
+    ),
+  ) {
+    this.storage = storage;
+    this.contractWrapper = contractWrapper;
+    this.metadata = new ContractMetadata(
```

Comment:
gonna have to figure out the flow of 
- detect interface
- instantiate corresponding 'feature' class

and apply that flow for all of these

Reply:
yup, will spend a bit of time on this today and improve it in this branch / get to something that could work

Reply:
yeah, give it another look, let me know if it's directionally more like what ur thinking, then I can unblock /react from it and u can take it over tomorrow from there?